types/util.go

```diff
@@ -31,7 +31,15 @@ var uintSizeTable = [21]uint64{
 
 // LenOfUint64Fast efficiently calculate the string character lengths of an uint64 as input
 func LenOfUint64Fast(x uint64) int {
-	for i := 1; ; i++ {
+	i := 1
+	// reduce comparison time for large numbers
+	if x > 999 {
+		i = 4
+	}
+	if x > 9999999 {
+		i = 8
+	}
```

Comment:
Do you benchmarked this? According to [this](https://gist.github.com/erjiaqing/53dd66c828ac1606cfef0c7f7d3e16fc) the speed might be a bit slower with random length.

```
goos: linux
goarch: amd64
BenchmarkGetLenDivide-8                 2000000000               1.42 ns/op
BenchmarkGetLenDivideBig-8              200000000                9.11 ns/op
BenchmarkGetLenLookUpTable-8            1000000000               2.81 ns/op
BenchmarkGetLenLookUpTableBig-8         1000000000               2.49 ns/op
BenchmarkGetLenLookUpTableBig2-8        2000000000               1.87 ns/op
PASS
```